src/components/forms/ChoiceList/ChoiceList.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import { Box } from '../../structure'
+import { Checkbox, CheckboxProps } from '../Checkbox/Checkbox'
+import { RadioButton, RadioButtonProps } from '../RadioButton/RadioButton'
+import { BodyText } from '../../text'
+import { InlineError } from '../InlineError/InlineError'
+
+export interface Choice {
+  /**
+   * Label for the choice.
+   */
+  label: React.ReactNode
+  /**
+   * Value of the choice.
+   */
+  value: string
+  /**
+   * Additional text to aid in use.
+   */
+  helpText?: React.ReactNode
+  /**
+   * Disable choice.
+   */
+  disabled?: boolean
+}
+
+export interface ChoiceListProps {
+  /**
+   * List of choices.
+   */
+  choices: Choice[]
+  /**
+   * List of selected choices.
+   */
+  selected: string[]
+  /**
+   * Label for the list of choices.
+   */
+  title: React.ReactNode
+  /**
+   * Allow multiple choices to be made.
+   */
+  allowMultiple?: boolean
+  /**
+   * Disable all choices.
+   */
+  disabled?: boolean
+  /**
+   * Display an error message.
+   */
+  error?: string
+  /**
+   * Callback when the selected choices change.
+   */
+  onChange: (selected: string[]) => void
+}
+
+/**
+ * Let you create a list of grouped radio buttons or checkboxes.
+ * Use this to group a list of choices together.
+ */
+export const ChoiceList: React.FC<ChoiceListProps> = ({
+  choices,
+  selected,
+  title,
+  allowMultiple = false,
+  error,
+  disabled = false,
+  onChange,
+}) => {
+  const itemProps: ItemsProps[] = choices.map(
+    ({ label, value, disabled: disabledItem, helpText }) => ({
+      key: value,
+      label,
+      checked: selected.indexOf(value) >= 0,
+      disabled: disabledItem || disabled,
```

Comment:
Why is this required?

Reply:
We want to disable an `Item` if the user expressly disabled it or it the whole list is disabled. This is why we have the OR condition.